var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -167,7 +167,8 @@ class Llvm(CMakePackage, CudaPackage):
     # Universal dependency
     depends_on("python@2.7:2.8", when="@:4+python")
     depends_on("python", when="@5:+python")
-    depends_on("z3", when="@9:")
+    # While not required for @8:9, cmake can find a z3 on the system, causing failures:
+    depends_on('z3', when='@8:')
```

Comment:
Could you add one thing here, since z3 is now on the required list back to 8, add the cmake define to use it: `LLVM_ENABLE_Z3_SOLVER`.  In principle it should be on if found, but if we're requiring it, I'd rather have it in the build so we find out if something goes wrong with the dependency.

Reply:
It is documented @ https://llvm.org/docs/CMake.html to be an option for the clang analyzer
Thus I turned it into a variant for people using the clang static analyzer and need it.

Two smaller fixes added as well:
* The `no_cyclades.patch` backported was included in `12.0.1`, so `12.0.0` still needs it.
* The backport of the ipc_perm patch for the sanitizer is also needed on Ubuntu `focal` with `gcc@9.3.0`